src/server/update_check.ts

```diff
@@ -0,0 +1,101 @@
+import { colors, join } from "./deps.ts";
+import { DEBUG } from "./constants.ts";
+
+interface CheckFile {
+  last_checked: string;
+  latest_version: string;
+  current_version: string;
+}
+
+async function getDenoDir(): Promise<string | undefined> {
+  const output = await new Deno.Command("deno", { args: ["info"] }).output();
+  const denoDir = colors.stripColor(new TextDecoder().decode(output.stdout))
+    .split(
+      /\n/g,
+    ).find(
+      (line) => line.startsWith("DENO_DIR"),
+    )?.replace("DENO_DIR location: ", "");
+
+  return denoDir;
+}
```

Comment:
This feels really bad. Spawning a subprocess to read the output text? I was looking at `/x/plug` since they also write stuff to the cache directory but they hardcode the path calculations depending on the OS.

Could we possibly use https://github.com/denoland/deno_cache instead?

Reply:
Agree, this was mostly the quickest way for the prototype, but yeah it's not ideal. Will switch to storing something in the user's home dir instead.